app/scripts/modules/core/src/managed/Environments.tsx

```diff
@@ -37,7 +38,14 @@ export function Environments({ app }: IEnvironmentsProps) {
   } = useDataSource(dataSource);
   const loadFailure = status === 'ERROR';
 
-  const [selectedArtifact, setSelectedArtifact] = useState<ISelectedArtifact>();
+  const {
+    stateService: { go },
+  } = useRouter();
+  const { params } = useCurrentStateAndParams();
+  const selectedArtifact = useMemo<ISelectedArtifact>(
+    () => (params.version ? pick(params, ['type', 'name', 'version']) : null),
+    [params.type, params.name, params.version],
```

Comment:
doing all this memoized `pick`ing looks and seems kind of gross, but I haven't thought of a better way to do this where:
1. `selectedArtifact` maintains reference equality until one of type/name/version changes.
2. we only get the fields we want out of `params` rather than passing the whole thing all over the place.